jdtls.ext/com.shengchen.checkstyle.checker/src/main/java/com/shengchen/checkstyle/checker/CheckerListener.java

```diff
@@ -40,7 +40,7 @@ public void addError(AuditEvent error) {
         }
         fileErrors.get(error.getFileName()).add(new CheckResult(
             error.getLine(),
-            error.getLocalizedMessage().getColumnCharIndex() + 1,
+            error.getViolation().getColumnCharIndex() + 1,
```

Comment:
The extension allows users to specify any version of Checkstyle. Is `getViolation()` exists in other versions of Checkstyle?

Maybe we need a try-catch fallback here I guess?

Reply:
No, `.getViolation()` is only available since v8.42. Unfortunately, this change will make versions below v8.42 incompatible with the plugin 😞

I'm all-in on trying to keep backward compatibility. However, Checkstyle's API changed on v8.42, meaning the code will not compile with both `.getLocalizedMessage()` and `.getViolation()`, so a try-catch may not work here.

An option could be to use Java 8 lookup methods (reflection) to check which method is available and invoke it conditionally. We'll lose a bit of type safety but keep backward compatibility. What do you think? 🙂

Reply:
Sure, let's use reflection to resolve this issue. If the declared method `getViolation()` cannot be get, we fall back to use `.getLocalizedMessage()`

Reply:
Sounds good to me! I updated the PR and confirmed that both `v8.41.1` and `v9,0` are working 🙂